examples/minelogy.py

```diff
@@ -181,31 +181,48 @@ def set_recipes(self, mcrecipes):
         self.crafts = []
         for mcrecipe in mcrecipes:
             craft_name = mcrecipe['name'].split('.')[-1]
-            craft_quantity = mcrecipe['count']
-            craft_tool = mcrecipe['recipe_type']
-            craft_group = mcrecipe['group'] # Need to understand how this could be used
-            ingredients = {}
-            craft_ingredients = []
-            for material in mcrecipe['ingredients']:
-                if len(material) == 0:
-                    continue
-                material_name = material[0]['type'].split(".")[-1]
-                if material_name not in ingredients:
-                    ingredients[material_name] = 1
-                else:
-                    ingredients[material_name] += 1
-            for ingredient in ingredients:
-                craft_ingredients.append({'type' : ingredient, 'quantity' : ingredients[ingredient]})
-            if len(craft_ingredients) == 0:
-                continue
-            additional_tool = self.add_craft_tool(craft_tool)
-            if additional_tool is not None:
-                craft_ingredients.extend(additional_tool)
-            craft_entity = (craft_ingredients,
-                    {'type': craft_name, 'quantity': craft_quantity})
-            if (craft_entity not in self.crafts) and (len(craft_entity) > 0):
+            craft_entity = self.set_one_recipe(mcrecipe, craft_name)
+            if craft_entity is not None:
                 self.crafts.append(craft_entity)
 
+    def set_recipes_for_items(self, items_to_add, mcrecipes, clear_recipes=False, strict_matching=True):  # here we're sending list of items we want to add recipes for
+        if clear_recipes:
+            self.crafts = []
+        for mcrecipe in mcrecipes:
+            craft_name = mcrecipe['name'].split('.')[-1]
+            if (craft_name in items_to_add) or (not strict_matching and craft_name.split("_")[-1] in items_to_add):
+                craft_entity = self.set_one_recipe(mcrecipe, craft_name)
+                if craft_entity is not None:
+                    self.crafts.append(craft_entity)
+
+    def set_one_recipe(self, mcrecipe, craft_name):
```

Comment:
Could you please add typing and/or a docstring? It's not clear where the recipe is to be set..